src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs

```diff
@@ -235,7 +235,11 @@ private void WriteAnalyzerFileReferenceMvid(AnalyzerFileReference reference, Obj
         {
             try
             {
-                using (var stream = new FileStream(reference.FullPath, FileMode.Open, FileAccess.Read, FileShare.Read | FileShare.Delete))
+                // use actual assembly path rather than one returned from reference.FullPath
+                // 2 can be different if analyzer loader used for the reference do something like shadow copying
```

Comment:
@heejaechang Can this be the root cause of https://github.com/dotnet/roslyn/issues/30890#issuecomment-434888485?